### PR TITLE
Persist configuration options to local storage

### PR DESCRIPTION
**Pull request by tbekolay**
The bulk of the logic is in `static/config.js`. So far we only persist the 4 configuration options accessible through the gear icon in the top toolbar, but we can add more things easily.

Addresses issue #375.

### CODE REVIEW

**Comment by tbekolay** [comment]
Updated this to also persist some options for the code editor.

The only thing that I think could persist that doesn't is the backend used... which I'm not sure it should be persisted. Thoughts?

**Comment by tcstewar** [comment]
> The only thing that I think could persist that doesn't is the backend used... which I'm not sure it should be persisted. Thoughts?

I'm okay with that not being persisted, especially since you can already set it on the command line...

**Comment by tbekolay** [comment]
Cool cool. I think maybe we should think about moving that somewhere other than through the Configuration Options modal window eventually, but definitely not a priority for now!

File: nengo_gui/static/ace.js [hunk 1]
```diff
@@ -174,68 +195,56 @@ Nengo.Ace.prototype.on_message = function (event) {
     }
 }
 
+Nengo.Ace.prototype.on_resize = function() {
+    this.max_width = $(window).width() - 100;
+    if (this.width > this.max_width) {
+        this.width = this.max_width;
+    }
+    console.log(this.max_width);
```
Comment (tcstewar) [inline nengo_gui/static/ace.js]: I assume this console.log should get removed

**Comment by tcstewar** [comment]
Very very nifty.  And other than the one console.log comment, I didn't find any other issues.  :)

File: nengo_gui/static/ace.js (hunk 1 above)
Comment (tbekolay) [inline nengo_gui/static/ace.js]: You are correct! I'll amend it.

**Comment by tbekolay** [comment]
console.log statement removed!

**Pull request merged**